### ISSUE TITLE
add public API to WebSocket to check if it's internal socketData->messageQueue is empty or not.

### ISSUE THREAD

**Issue by AlexVK79** [uWebSockets/uWebSockets#182]
Suppose our game server updates game logics 30/60 frames per seconds and sends 
game state messages to each client. Now if some client is a mobile device on a slow 3g or even 2g connection 
it could happen that it can't consume the stream we are sending him with the same speed as we send. 

It would not be a problem with UDP, but with uWebSocket implementation it might probably (theoretically) 
lead to the internal messageQueue grow until server crashes due to not enough RAM and/or that mobile client 
just get the increasing and increasing latency because of consumnig all that old game states queued.

The obvious solution is to not send anything to the client until we get some ACK from him
about some previous game state(s) received. But WebSocket and TCP/IP already have their
own acknowledgements and with the additional ACK "give me next state" we would just increase
latency for the clients that have no problems with consuming server's stream of game states as is.

So basically if we could do:

if (m_socket.isSendQueueEmpty())
	m_socket.send((char *)pGameState, size, OpCode::BINARY);

instead of just:

m_socket.send((char *)pGameState, size, OpCode::BINARY);

it would be just great.

**Comment by alexhultman** [comment]
You pass a callback to the WebSocket::send and this gets called when this message gets written to kernel buffer. From this you can infer the size of the queue.

**Comment by AlexVK79** [comment]
Thanks! That does the job! Although considering TCP send buffer size in OS, this was a wrong approach to begin with ))
